nginx.md

```diff
@@ -0,0 +1,190 @@
+	root /var/www/example;
+
+	index index.php;
+}
+```
+
+Let's break down this example.  The first line, `listen 80` tells Nginx to listen on port 80 for incoming connections.  This is the default port used by web traffic, and what you'll want to use 99% of the time.  `server_name` is where you specify what domain name or names match this configuration.  You'll likely want to include your domain both with and without the `www.` subdomain.  `root` lets you specify the Document Root for this site.  This is usually going to be where you've installed Pico, but ultimately it depends on your configuration.  `index index.php` tells Nginx that your site's index file will be called `index.php`.  This is necessary for Pico, however you can use something like `index index.html index.htm index.php` if you'd like Nginx to also search for a non-Pico html index file.
+
+Below we'll add a few more sections to this server configuration.  All of the following examples should be placed **inside** the server block, before the closing `}`
+
+## Denying Access to Pico's Internal Files
+
+One of the important features provided by Pico's `.htaccess` is the denial of access to some of Pico's directories.  There are two simple reasons for this, added security, and ease-of-use.  We don't want anyone snooping around and reading files they shouldn't be (like our Pico `config.php`), but we also don't want people accidentally navigating to a raw markdown file or viewing directory indexes (not actually an issue, since they're disabled unless specified in Nginx, but this isn't always the case in Apache).
+
+You should always use the following rule in the Nginx config of your Pico site:
+
+```
+location ~ /(\.htaccess|\.git|config|content|content-sample|lib|vendor) {
+	return 404;
+}
+```
+
+This rule returns a 404 (file not found) error if someone tries to access `.htaccess` or Pico's other internal files.  Remember, `.htaccess` is a configuration file for Apache, and Apache would have denied access to it automatically.  While it doesn't do anything under Nginx, you may have unknowingly left it in your Pico directory, as it is hidden by default.  While you can safely remove `.htaccess` from your Pico installation, we'd recommend you keep it in this rule anyway just in case.  For example, you may upgrade Pico in the future and forget to remove it!  In general though, this rule is for security, and a similar rule is usually recommend in Nginx, even outside of running Pico.
+
+The rest of the rule returns a 404 page if the user tries to navigate into Pico's internal file structure.  We recommend this as it's generally a good practice.  Users's don't need access to these files, so why allow it?
+
+## PHP Configuration
+
+This is a topic outside the realm of this document.  Unlike Apache (which sends every document to PHP automatically), Nginx is more streamlined.  It needs to be *told* to send a file to an external PHP processor.  It does not handle PHP itself.
```

Comment:
> Unlike Apache (which sends every document to PHP automatically), Nginx is more streamlined.

That's not entirely true. :wink: Actually Apache and Nginx handle PHP the exact same way: you configure a file extension (or multiple file extensions) and the webserver sends all requests to files with this file extension to PHP (still abridged for Apache; Apache uses MIME types, but they ultimately also mostly depend on file extensions). The difference between `mod_php` (Apache) and FastCGI (both Apache and Nginx) is about how the webserver *communicates* with PHP (i.e. how PHP processes are handled, so it's a more technical question), not how requests are handled. The difference between Apache and Nginx is that Linux distributions ship more intelligent default configurations for Apache, so Apache users don't have to deal with this on their own.